spoon-control-flow/src/main/java/fr/inria/controlflow/ControlFlowBuilder.java

```diff
@@ -798,7 +798,11 @@ public <T, S> void visitCtSwitchExpression(CtSwitchExpression<T, S> switchExpres
 
 	@Override
 	public void visitCtSynchronized(CtSynchronized synchro) {
+		ControlFlowNode expressionNode = new ControlFlowNode(synchro.getExpression(), result, BranchKind.STATEMENT);
+		tryAddEdge(lastNode, expressionNode);
+		lastNode = expressionNode;
```

Comment:
Do we want to preserve the synchronized statement itself in the graph? I.e. call `defaultAction(synchro, BranchKind.STATEMENT)` or w/e somewhere? Currently it looks like the synchronized is unrolled and no trace of it is left?

Reply:
Ideally yes, but in the current CFG model this is not possible in any way I like. This is one of the things I would like to improve with the redesign started in #5742